Power_Of_Any_Number/PowerOfAnyGivenNo.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Problem Statement
+ * 	Program to find if a No. is a power of any Given No. in Java
+ *	
+ *Approach
+ *	comparing the log's value with the floor value of log 
+ */
+import java.util.Scanner;
+
+public class PowerOfAnyGivenNo {
+
+	public static void main(String[] args) {
+		Scanner sc = new Scanner(System.in);	//input
+		System.out.print("Enter a no. : ");	
+		int number = sc.nextInt();
+		
+		System.out.print("Check for which no. ? ");	//base element
+		int base = sc.nextInt();
+		
+		if( isPowerOfGivenNo(number, base) )
+			System.out.println("Yes "+number +" is a power of "+base);
+		else
+			System.out.println("No "+number +" is not a power of "+base);
+		sc.close();
+	}
+
+	private static boolean isPowerOfGivenNo(int number, int base) {
+		
+		//explicitly check for 1 because log(1) = 0
+		if(base == 1){
+			if(number == 1)
+				return true;
+			else	
+				return false;
+		}
+		//converting to positive	
+		number = Math.abs(number);
```

Comment:
Are you handling -ve scenarios ?
Because it looks like your code will say "-2 is a power of 2", however it is not. 
Please can you verify and handle the cases?

Reply:
oh, my bad, will add code to deal with -ve scenarios!

Reply:
Added additional cases to handle situations like these